src/fmt/writer/termcolor/extern_impl.rs

```diff
@@ -70,46 +71,71 @@ impl Formatter {
 
 pub(in crate::fmt::writer) struct BufferWriter {
     inner: termcolor::BufferWriter,
-    test_target: Option<Target>,
+    test_target: Option<WritableTarget>,
 }
 
 pub(in crate::fmt) struct Buffer {
     inner: termcolor::Buffer,
-    test_target: Option<Target>,
+    test_target: bool,
```

Comment:
This change I'm not so sure about. I haven't really tried to understand the surrounding code, but it was the easiest fix.

I am confident that the behavior is still the same as before, but maybe there was some reason the test target was being stored before? Also, having the same field name as in `BufferWriter` but a different type seems like a code smell, but I couldn't really come up with something better due to lack of understanding of the code overall.

Reply:
At first I was a bit confused, but I can not find anything wrong and I think it's an improvement.
Maybe it would be less confusing to rename `Buffer::test_target` to `Buffer::is_test`?

Reply:
I'll check again how it's used and rename either to `is_test` or sth. like `has_test_target`.

Reply:
Ye, I do agree on the `has_test_target`.